src/qat/purr/integrations/grammars/stdgates.inc

```diff
@@ -0,0 +1,155 @@
+// OpenQASM 3.0 standard gate library
+//
+// Vendored from the OpenQASM 3.0 project at commit 4ca1d793833b24a1 of
+// https://github.com/openqasm/openqasm.
+//
+// Used under the terms of its Apache-2.0 license.
+// Copyright 2017-2024 OpenQASM Contributors.
+//
+// This version has been altered by Oxford Quantum Circuits
+
+
+// phase gate
+// gate p(λ) a { ctrl @ gphase(λ) a; }
+gate p(λ) a { u1(λ) a; }
+
+// Pauli gate: bit-flip or NOT gate
+gate x a { U(π, 0, π) a; gphase(-π/2);}
+// Pauli gate: bit and phase flip
+gate y a { U(π, π/2, π/2) a; gphase(-π/2);}
+ // Pauli gate: phase flip
+gate z a { p(π) a; }
+
+// Clifford gate: Hadamard
+gate h a { U(π/2, 0, π) a; gphase(-π/4);}
+// Clifford gate: sqrt(Z) or S gate
+// gate s a { pow(0.5) @ z a; }
+gate s a { u1(pi/2) a; }
+// Clifford gate: inverse of sqrt(Z)
+// gate sdg a { inv @ pow(0.5) @ z a; }
+gate sdg a { u1(-pi/2) a; }
+
+// sqrt(S) or T gate
+// gate t a { pow(0.5) @ s a; }
+gate t a { u1(pi/4) a; }
+// inverse of sqrt(S)
+// gate tdg a { inv @ pow(0.5) @ s a; }
+gate tdg a { u1(-pi/4) a; }
+
+// sqrt(NOT) gate
+// gate sx a { pow(0.5) @ x a; }
+gate sx a { sdg a; h a; sdg a; }
```

Comment:
where is this re-defintion taken from?

Reply:
All the re-definitions are taken from the QASM2 version: https://github.com/Qiskit/qiskit/blob/main/qiskit/qasm/libs/qelib1.inc